### PR TITLE
[#3966] feat(core): add cloud and region property for catalog

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

add `cloud.name` and `cloud.region-code` property for catalog

### Why are the changes needed?

Fix: #3966 

### Does this PR introduce _any_ user-facing change?

yes, introduce some limitations of `cloud.name` property

### How was this patch tested?

tests added

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/connector/BaseCatalogPropertiesMetadata.java [hunk 1]
```diff
@@ -15,6 +15,9 @@
 @Evolving
 public abstract class BaseCatalogPropertiesMetadata extends BasePropertiesMetadata {
 
+  public static final String CLOUD_NAME = "cloud.name";
+  public static final String CLOUD_REGION_CODE = "cloud.region-code";
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/connector/BaseCatalogPropertiesMetadata.java:19]: It is better to define these two reserved properties in Catalog interface. So users will know how to use it. Also don't forget to update the doc.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 2]
```diff
@@ -27,6 +27,15 @@ To use fileset, please make sure that:
...
  - A metalake has been created.
 
+## Catalog properties
+
+The following are the common properties for all fileset catalogs:
+
+| Property Name       | Description                                                                                                                             | Default Value | Required | Since Version |
+|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------|----------|---------------|
+| `cloud.name`        | The property to specify the cloud that the catalog is running on. The valid values are `aws`, `azure`, `gcp`, `on_premise` and `other`. | (none)        | No       | 0.6.0         |
+| `cloud.region-code` | The property to specify the region code of the cloud that the caatlog is running on.                                                    | (none)        | No       | 0.6.0         |
```
Comment (jerryshao) [inline docs/manage-fileset-metadata-using-gravitino.md:37]: There's a chapter in `gravitino-server-config.md` describing about the common catalog properties, maybe you should combine them together.

image

Comment (jerryshao) [reply docs/manage-fileset-metadata-using-gravitino.md:37]: Maybe you can also move to here.

Comment (mchades) [reply docs/manage-fileset-metadata-using-gravitino.md]: fixed

**Review by jerryshao — APPROVED** [review]

**Pull request merged**